### ISSUE TITLE
FP not working on OnePlus 6T

### ISSUE THREAD

**Issue by Zyonse** [phhusson/treble_experimentations#449]
The (under display) fingerprint sensor doesn't work at all on the OnePlus 6T. Logcat attached below was taken on the enrollment screen.

[enroll_aosp.txt](https://github.com/phhusson/treble_experimentations/files/3015942/enroll_aosp.txt)

**Comment by penn5** [comment]
This is a known issue, we discussed it a few months ago in telegram, but the HAL is not very nice and phh didn't get very far, iirc.

**Comment by phhusson** [comment]
Yeah I've started working on this, though I don't have an OP6T so that's a bit complicated.

My current knowledge is described here:
https://github.com/phhusson/vendor_interfaces/tree/master/vendor/oneplus/fingerprint/extension/1.0

Additionnaly to that, I fixed under-display fingerprint sensor on Mi 9, so I now understand how such sensors work.

Though it seems OP did really weird thing in kernel to support it, which may actually make things harder to develop.

**Comment by phhusson** [comment]
Okay, some additions.

Looks like drm driver notifies fingerprint driver which notifies fingerprint daemon about when finger is pressed/green dot is displayed.

Technically we could talk directly to fingerprint daemon (it is using NETLINK sockets), but OP decided their HAL would be drm /sys entries. (that's stupid but whatever). We can see that because their are two under-display fingerprint drivers (silead & goodix) in the kernel, with different netlink magics.

So our entrypoints are all in /sys' drm:
- op_friginer_print_hbm
- force_screenfp
- notify_fppress
- notify_dim

**Comment by Austin416** [comment]
FingerPrint still not working

**Comment by VinTz1** [comment]
Hello, do you think this bug and FP not working on OnePlus 7 are the same?

**Comment by phhusson** [comment]
I'm guessing Oneplus has the same framework code for those devices, yes
